blockchains/eth/lib/fees_decoder.js

```diff
@@ -12,52 +11,114 @@ class FeesDecoder {
       from: transaction.from,
       to: block.miner,
       value: computeGasExpense(this.web3Wrapper, transaction.gasPrice, receipts[transaction.hash].gasUsed),
-      valueExactBase36: computeGasExpenseBase36(this.web3, transaction.gasPrice, receipts[transaction.hash].gasUsed),
+      valueExactBase36: computeGasExpenseBase36(this.web3Wrapper, transaction.gasPrice, receipts[transaction.hash].gasUsed),
       blockNumber: this.web3Wrapper.parseHexToNumber(transaction.blockNumber),
       timestamp: this.web3Wrapper.parseHexToNumber(block.timestamp),
       transactionHash: transaction.hash,
       type: "fee"
     }]
   }
 
-  getPostLondonForkFees(transaction, block, receipts) {
-    const result = []
-    const maxPriorityFeePerGas = transaction['maxPriorityFeePerGas']
+  pushBurntFee(transaction, block, receipts, result) {
+    result.push({
+      from: transaction.from,
+      to: constants.BURN_ADDRESS,
+      value: computeGasExpense(this.web3Wrapper, block.baseFeePerGas, receipts[transaction.hash].gasUsed),
+      valueExactBase36: computeGasExpenseBase36(this.web3Wrapper, block.baseFeePerGas,
+        receipts[transaction.hash].gasUsed),
+      blockNumber: this.web3Wrapper.parseHexToNumber(transaction.blockNumber),
+      timestamp: this.web3Wrapper.parseHexToNumber(block.timestamp),
+      transactionHash: transaction.hash,
+      type: "fee_burnt"
+    })
+  }
 
-    if (this.web3Wrapper.parseValue(maxPriorityFeePerGas) > 0) {
+  pushType2MinerFee(transaction, block, receipts, result) {
+    /***
+     * An EIP1559 transaction always pays the base fee of the block it’s included in, and it pays a priority fee
+     * as priced by maxPriorityFeePerGas or, if the base fee per gas + maxPriorityFeePerGas exceeds maxFeePerGas,
+     * it pays a priority fee as priced by maxFeePerGas minus the base fee per gas.
+     *
+     * EIP1559 transactions must specify both maxPriorityFeePerGas and maxFeePerGas. They must not specify gasPrice.
+     *
+     * https://besu.hyperledger.org/en/stable/Concepts/Transactions/Transaction-Types/
+     */
+    const maxPriorityFeePerGas = this.web3Wrapper.parseValueToBN(transaction['maxPriorityFeePerGas'])
+    const maxFeePerGas = this.web3Wrapper.parseValueToBN(transaction['maxFeePerGas'])
+    const baseFeePerGas = this.web3Wrapper.parseValueToBN(block.baseFeePerGas)
+
+    const minerFeePerGas = baseFeePerGas.add(maxPriorityFeePerGas).gt(maxFeePerGas) ?
```

Comment:
`transaction['gasPrice'] - block.baseFeePerGas`

When I query mined tx type 2 I got field gasPrice, isn't this the same?

Reply:
Thanks for this observation. I did not read the whole EIP and thought that the 'gasPrice' field is a leftover to be removed. You are right that it must be present and we can depend on it.